src/main/org/firebirdsql/gds/ng/wire/WireConnection.java

```diff
@@ -275,6 +300,14 @@ public final C identify() throws SQLException {
                     protocolVersion = (protocolVersion & FB_PROTOCOL_MASK) | FB_PROTOCOL_FLAG;
                 }
 
+                if (op_code == op_cond_accept || op_code == op_accept_data) {
+                    String pluginName = xdrIn.readString(getEncoding());
+                    final int is_authenticated = xdrIn.readInt();
+                    if (pluginName == "Legacy_Auth" && is_authenticated==0) {
+                        throw new SQLException("Unauthorized");
```

Comment:
Throw `335544472=Your user name and password are not defined. Ask your database administrator to set up a Firebird login.` instead?